src/app/api/v1/transaction/route.ts

```diff
@@ -0,0 +1,121 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { validatePublicKeyString } from '@/utils/publicKey';
+import { validate, v4 as uuidv4 } from 'uuid';
+import { createTransaction, getTransaction } from '@/logic/transactionLogic';
+import { Transaction, TransactionStatus, transactionStatuses } from '@/app/types/transaction';
+import { supportedMints } from '@/app/config/mint';
+
+// This type is used to return the transaction to the client.
+// It is used to hide certain fields from the public.
+export type PublicTransaction = Omit<Transaction, 'signedTransactionBytes' | 'requestedByIp'> & {
+  currentStatus: TransactionStatus;
+};
+
+// Convert a Transaction object to a PublicTransaction object
+export const transactionToPublicTransaction = (transaction: Transaction): PublicTransaction => {
+  return {
+    ...transaction,
+    currentStatus: transaction.currentStatus as TransactionStatus,
+  };
+};
+
+// Validate the request body for creating a new transaction
+const validateCreateTransactionRequest = (req: CreateTransactionRequest): { error: string } | null => {
+  const { referenceId, amount, mintSymbol, destination, sender } = req.body;
+  if (!validatePublicKeyString(destination)) {
+    return { error: 'Invalid destination public key' };
+  }
+  if (!validatePublicKeyString(sender)) {
+    return { error: 'Invalid sender public key' };
+  }
+  // referenceId is optional but should be an alphanumeric string
+  if (referenceId && !/^[a-zA-Z0-9]+$/.test(referenceId)) {
+    return { error: 'Invalid reference ID' };
+  }
+  // amount should be a float string
+  if (typeof amount !== 'string' || isNaN(Number(amount))) {
+    return { error: 'Invalid amount' };
+  }
+  if (!mintSymbol || typeof mintSymbol !== 'string' || !(mintSymbol in supportedMints)) {
+    return { error: 'Unsupported token' };
+  }
+
+  return null;
+}
+
+// Convert the request body to a Transaction object
+const convertCreateTransactionRequestToTransaction = (req: CreateTransactionRequest): Transaction => {
+  const { referenceId, amount, mintSymbol, destination, sender } = req.body;
+  const mint = supportedMints[mintSymbol as keyof typeof supportedMints];
+  const id = uuidv4();
```

Comment:
I might change this to v7 in the morning: v7 is sortable by unix time.. don't think it really matters though since every transaction will have a created at field